ruby/rails/security/audit/xss/avoid-session-manipulation.yaml

```diff
@@ -0,0 +1,21 @@
+rules:
+- id: avoid-session-manipulation
+  metadata:
+    owasp: 'A6: Security Misconfiguration'
+    cwe: 'CWE-276: Incorrect Default Permissions'
+    references:
+    - https://brakemanscanner.org/docs/warning_types/session_manipulation/
+    category: security
+    technology:
+    - rails
+  message: |
+    [Sorry I'm really confused about why this is actually a vulnerability. It looks like a read, but the description sounds like it's a write]
```

Comment:
Very confused

Reply:
it is kinda both read an write

```ruby
user_id = session[params[:name]]
current_user = User.find(user_id)
```
in this example user_id is expected to be retrieved form session, lets say it is expected that `:name` will have a value smth like `"user_id"`, but bc user can control it, there is an opportunity to retrieve smth else e.g. `"_csrf_token"` of `"username"` etc. But the data will be treated as `user_id` and it can lead to unexpected behavior

Reply:
so it is a good rule! just need a message for it :+1: